ckanext/xloader/views.py

```diff
@@ -21,3 +23,36 @@ def resource_data(id, resource_id):
         except ValueError:
             rows = None
     return utils.resource_data(id, resource_id, rows)
+
+
+@xloader.route("/dataset/<id>/delete-datastore/<resource_id>", methods=("GET", "POST"))
+def delete_datastore_table(id, resource_id):
```

Comment:
Why does this need the package ID?

Reply:
its to continue to match the comment pattern on all other queries.

Reply:
Fair enough. I would suggest, then, that there should be a sanity check to ensure that the resource actually belongs to the specified package.

Reply:
Okay can add that check, the package id is also needed for the redirect to `xloader.resource_data`

Reply:
Ah right, missed that one.

Reply:
@ThrawnCA okay, now asserting the resource ownership